src/arena/arena_loader.py

```diff
@@ -28,21 +29,71 @@ def load_arena_by_name(name: str) -> Arena:
                 )
-            converted_position = (
-                object.coordinates.x,
-                y_offset - object.coordinates.y,
-            )
-            rotation = radians(-object.rotation)
-            center_offset = scale_vec((object.size.width, -object.size.height), 0.5)
-            rotated_center_offset = rotate_vec(center_offset, rotation)
-            center = add_vec(rotated_center_offset, converted_position)
-
-            wall = Wall(
-                int(center[0]),
-                int(center[1]),
-                int(object.size.width),
-                int(object.size.height),
-                rotation,
-            )
+            transform, size = get_rectangle_object_transform_and_size(object)
+
+            wall = Wall(transform, size)
             arena._walls.append(wall)
+        elif object.class_ == "SpawnPoint":
+            transform, size = get_tile_object_transform_and_size(object)
+            initial_spawn_for_player_str = object.properties.get(
+                "initial_spawn_for_player"
+            )
+            initial_spawn_for_player = None
+            if initial_spawn_for_player_str is not None:
+                initial_spawn_for_player = int(initial_spawn_for_player_str)
+
+            spawn_point = SpawnPoint(transform, initial_spawn_for_player)
+
+            arena._spawn_points.append(spawn_point)
+            if initial_spawn_for_player is not None:
+                arena._initial_spawn_points[initial_spawn_for_player] = spawn_point
+
     return arena
+
+
+def get_tile_object_transform_and_size(object: TiledObject):
+    """
+    Return the transform and dimensions of a Tile Object loaded from Tiled map editor.
+
+    Returns (transform, size) where:
+
+    transform = (x, y, radians)
+
+    size = (width, height)
+    """
+    return get_object_transform_and_size(object, True)
+
+
+def get_rectangle_object_transform_and_size(object: TiledObject):
+    """
+    Return the transform and dimensions of a Rectangle Object loaded from Tiled map editor.
+
+    Returns (transform, size) where:
+
+    transform = (x, y, radians)
+
+    size = (width, height)
+    """
+    return get_object_transform_and_size(object, False)
+
+
+def get_object_transform_and_size(object: TiledObject, pivot_is_at_bottom: bool):
+    y_offset = SCREEN_HEIGHT
+    converted_position = (
+        object.coordinates.x,
+        y_offset - object.coordinates.y,
+    )
+    rotation = radians(-object.rotation)
+
+    # TODO I kinda hate this syntax; why is python like this?
+    height_offset = object.size.height if pivot_is_at_bottom else -object.size.height
+
+    center_offset = scale_vec((object.size.width, height_offset), 0.5)
+    rotated_center_offset = rotate_vec(center_offset, rotation)
+    center = add_vec(rotated_center_offset, converted_position)
+
+    transform = (center[0], center[1], rotation)
+
+    size = (object.size.width, object.size.height)
+
+    return (transform, size)
```

Comment:
Non blocking: Would this kind of function eventually make its way to iron_math, or is it too specific to arena objects

Reply:
It's pretty specific to the Tiled map editor since it accepts `TiledObject`.  If we start loading non-arena stuff from Tiled, then we could move this into a `tiled.py` or `map_editor.py` module of helpers for loading maps.

I can imagine the map editor being used for visual layout of other stuff, perhaps a main menu, character select screen.  Those screens would still be driven by plenty of code, but an artistic touch to position everything would help, and the map editor could allow a non-coder to do that.  Mockup some pngs of the visual elements, drag-and-drop them into the map editor, and then our code can extract coordinates.

We might also split `iron_math` into two files: one for the pure math stuff, and the other for all the sprite movement routines.